servers/rendering/renderer_canvas_cull.cpp

```diff
@@ -247,17 +248,24 @@ void RendererCanvasCull::_cull_canvas_item(Item *p_canvas_item, const Transform2
 		}
 	}
 
-	Transform2D xform = ci->xform;
+	Transform2D final_xform;
+	if (!_interpolation_data.interpolation_enabled || !ci->interpolated) {
+		final_xform = ci->xform_curr;
+	} else {
+		real_t f = Engine::get_singleton()->get_physics_interpolation_fraction();
+		TransformInterpolator::interpolate_transform_2d(ci->xform_prev, ci->xform_curr, final_xform, f);
+	}
+
 	Transform2D parent_xform = p_parent_xform;
 
 	if (snapping_2d_transforms_to_pixel) {
-		xform.columns[2] = xform.columns[2].round();
+		final_xform.columns[2] = final_xform.columns[2].round();
 		parent_xform.columns[2] = parent_xform.columns[2].round();
```

Comment:
Note: If you are doing snapping here, you may want the snapping to apply to the local object, but NOT to propagate to the children, so they can do their snapping independently. This may entail creating an extra intermediate e.g. `snapped_xform` and use this for rendering, while passing the unchanged `final_xform` to the children.

This certainly needs consideration because snapping is done differently in 3.x.

Reply:
I tried multiple times snap the children independently (ask @markdibarry) but it always resulted with snapping issues. I think it's best to snap the parent as is.